lib/middleware/validate-signature-middleware.js

```diff
@@ -0,0 +1,267 @@
+            currentDate = OtDate.now();
+
+            // Allow a window of `lifetime` around currentDate.  We can
+            // test for this by ensuring these two things:
+            //
+            //     requestDate < currentDate + dateWindowFuture
+            //         Otherwise it is too far in the future
+            //     requestDate > currentDate - dateWindowPast
+            //         Otherwise it is too far in the past
+            //
+            // To make it work with OtDate, we change the second condition
+            // to look like this:
+            //
+            //     requestDate + dateWindowPast > currentDate
+            if (currentDate.plus(dateWindowFuture).isBefore(requestDate)) {
+                return errorResponse.rejectedPromiseAsync("The X-OpenToken-Date header is in the future.", "uYom0uhM");
+            }
+
+            if (requestDate.plus(dateWindowPast).isBefore(currentDate)) {
+                return errorResponse.rejectedPromiseAsync("The X-OpenToken-Date header is too far in the past.", "JzFzaWPt");
+            }
+
+            return true;
+        });
+    }
+
+
+    /**
+     * Compares a message signature against the request.  Fails if the
+     * signature is invalid.  If `allowUnsigned` is not truthy, fails if
+     * there is no Authorized header.
+     *
+     * @param {Restify~Request} req
+     * @param {boolean} allowUnsigned
+     * @return {Promise.<boolean>} true if this is a signed request
+     */
+    function authenticateAsync(req, allowUnsigned) {
+        // Take care of requests without the Authorized header
+        if (!req.headers.authorized) {
+            if (allowUnsigned) {
+                return promise.resolve(false);
+            }
+
+            return errorResponse.rejectedPromiseAsync("A signed message is required.", "0FLNjTwn");
+        }
+
+        // The header exists.  Authenticate the Host header, the
+        // X-OpenToken-Date header and finally the signature.
+        return verifyHostHeaderAsync(req.headers.host).then(() => {
+            return verifyXOpenTokenDateHeaderAsync(req.headers["x-opentoken-date"]);
+        }).then(() => {
+            return authenticateSignatureAsync(req, req.headers.authorized);
+        }).then(() => {
+            // Indicate that this is a signed request.
+            return true;
+        });
+    }
+
+
+    dateWindowFuture = config.signature.dateWindowFuture;
+    dateWindowPast = config.signature.dateWindowPast;
+    signatureHost = config.signature.host;
+    signatureMethod = config.signature.method;
+    signatureLinkDefinition = {
+        related: {
+            href: config.signature.relatedLink,
+            title: "signature-information"
+        }
+    };
+
+    /**
+     * Middleware that calls authenticate and then either signals that
+     * the request is signed or else indicates the error and prevents
+     * future middleware from running.
+     *
+     * @param {boolean} [allowUnsigned=false]
```

Comment:
You use brackets when a parameter is optional.  If it is optional and defaults to a specific value then you can make that very explicit by documenting it.  Normally I don't care too much for things like `options`, but this one is special and I want to be very clear that if you don't pass `true` then unsigned requests are *not* allowed.